corehq/apps/translations/app_translations/upload_app.py

```diff
@@ -247,35 +258,22 @@ def update(self, rows):
         """
         self.msgs = []
         for row in get_unicode_dicts(rows):
-            identifying_text = row.get('menu_or_form', row.get('sheet_name', ''))
-            identifying_parts = identifying_text.split('_')
-
-            if len(identifying_parts) not in (1, 2):
-                self.msgs.append((
-                    messages.error,
-                    _('Did not recognize "%s", skipping row.') % identifying_text
-                ))
-                continue
+            sheet_name = row.get('menu_or_form', row.get('sheet_name', ''))
+            unique_id = row.get('unique_id')
+
+            if unique_id and sheet_name not in self.names_map:
+                self.names_map[sheet_name] = unique_id
+            elif not unique_id and sheet_name in self.names_map:
+                unique_id = self.names_map[sheet_name]
```

Comment:
can you add a quick note on why we are checking on unique_id column here when we already have the names_map?

Reply:
Yes, a comment would definitely make a difference here. (The "unique_id" column is not always populated. So line 265 builds `names_map` when we have a value for `unique_id`, so we can fetch it from `names_map` here when we don't have a value for it.)

Reply:
its not clear why you would not have a unique_id available earlier. may be i just need to read better.